File: leetcode/3516/c/main.c
#include <math.h>
#include <stdlib.h>

int findClosest(int x, int y, int z) {
    int distance1 = abs(z - x);
    int distance2 = abs(z - y);

    if (distance1 < distance2) {
        return 1;
    } else if (distance1 > distance2) {
        return 2;
    } else {
        return 0;
    }
}

File: leetcode/3516/rs/src/main.rs
struct Solution;
impl Solution {
    pub fn find_closest(x: i32, y: i32, z: i32) -> i32 {
        match (z - x).abs().cmp(&(z-y).abs()) {
            std::cmp::Ordering::Less => 1,
            std::cmp::Ordering::Greater => 2,
            std::cmp::Ordering::Equal => 0,
        }
    }
}

fn main() {
    todo!()
}
